chan.go

```diff
@@ -70,6 +82,20 @@ func (r *chanResponse) Length() uint64 {
 	return r.length
 }
 
+func (re *chanResponse) Head() Head {
```

Comment:
Whats this method supposed to do?

Reply:
It's not new, I just moved it from here: https://github.com/ipfs/go-ipfs-cmds/pull/112/files/15c1508126b5934ffae976e4a09c4353fa90956e#diff-d1319de8df19ba21c67a62fa1fde5f89L201

But I think it's needed for HEAD requests, it's even in the ResponseEmitter interface; but it looks like it's not called from anywhere in `go-ipfs-cmds/...`, so maybe we can drop it. I'd suggest doing that in a new PR though.

Reply:
I decided to do it in this PR.